Northeastern-Electric-Racing/Siren
# Initial NanoMQ/EMQX configuration

## Changes

Adds basic EMQX and NanoMQ conf files.  Includes sqlite backup on TPU, bridging with the IP addresses of TPU and base station, etc.  Does not include acl.  Also does not include way for one to subscribe to # on EMQX, that can be added later as part of emqx acl.

Nanomq remains masked at startup.

## Notes

Below summary of nanomq issues and possible resolutions:
- Sqlite persistence failure (workaround applied in PR: optimziations disabled)
-  #78 (workaround applied in prev. PR: assign nanomq PID with start-stop-daemon)
- #79 (caused by hostapd, more investigation needed)**
- Nanomq dies when bridge not up (no workaround, fixed upstream and awaiting release)
- EMQX soft crashes when MQTTX is used to susbcribe to the same topic from both sides of the bridge (no workaround, issue not a big deal)

## Test Cases

- Send any topic from nanomq to emqx
- Observe messages sent over bridge
- Observe sqlite db logging all messages

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please reach out to your Project Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [n/a] All checks passing

Closes #76
